source/vir/src/recursive_types.rs

```diff
@@ -722,6 +722,21 @@ pub fn check_traits(krate: &Krate, ctx: &GlobalCtx) -> Result<(), VirErr> {
     // This also ensures that whenever A is used in f and g,
     // the dictionary a: Dictionary_U<A> is available.
 
+    // To handle bounds on trait methods like this:
+    //   trait T {
+    //     fn f<A: U>(x: Self, y: Self) -> bool;
+    //     fn g(x: Self, y: Self) -> Self { requires(f(x, y)); };
+    //   }
+    // We take a Dictionary_U as a parameter:
+    //   struct Dictionary_T<Self, A> {
+    //     f: Fn<A>(a: Dictionary_U<A>, x: Self, y: Self) -> bool,
+    //     g: Fn(x: Self, y: Self) -> Self { requires(f(x, y)); },
+    //   }
+    // This adds an edge:
+    //   - T --> U
```

Comment:
Where is this already handled?

Reply:
I just pushed an update that clarifies this comment.  In `expand_call_graph`, the `Add f --> T for any function f with "where ...: T(...)"` creates `f --> U`.